src/WebJobs.Script.WebHost/Management/InstanceManager.cs

```diff
@@ -122,16 +124,34 @@ public bool StartAssignment(HostAssignmentContext context)
             }
         }
 
-        public async Task<string> ValidateContext(HostAssignmentContext assignmentContext)
+        public Task<string> ValidateContext(HostAssignmentContext assignmentContext)
         {
             _logger.LogInformation($"Validating host assignment context (SiteId: {assignmentContext.SiteId}, SiteName: '{assignmentContext.SiteName}')");
+            // In AppService, ZipUrl == 1 means the package is hosted in azure files.
+            // Otherwise we expect zipUrl to be a blobUri to a zip or a squashfs image
+            if (!string.IsNullOrEmpty(assignmentContext.ZipUrl) && assignmentContext.ZipUrl != "1")
+            {
+                return ValidateBlobPackageContext(assignmentContext.ZipUrl);
+            }
+            else if (!string.IsNullOrEmpty(assignmentContext.AzureFilesConnectionString))
+            {
+                return ValidateAzureFilesContext(assignmentContext.AzureFilesConnectionString, assignmentContext.AzureFilesContentShare);
+            }
+            else
+            {
+                var error = $"Missing ZipUrl and AzureFiles. {nameof(ValidateContext)}";
```

Comment:
Because checking the assignment context makes the function host fail early.
The test **Assign_MSISpecializationFailure_ReturnsError** doesn't set RUN_FROM_PACKAGE so it gets 400 even before the MSI assignment check.
The test **ValidateContext_EmptyZipUrl_ReturnsSuccess** expects the error to be null but here we return an error message.